File: src/main/java/top/criwits/scaleda/verilog/psi/factory/nodes/ParameterDeclarationPsiNode.java
package top.criwits.scaleda.verilog.psi.factory.nodes;

import com.intellij.lang.ASTNode;
import top.criwits.scaleda.verilog.psi.TypedDeclaration;
import org.antlr.intellij.adaptor.psi.ANTLRPsiNode;
import org.jetbrains.annotations.NotNull;

public class ParameterDeclarationPsiNode extends ANTLRPsiNode
        implements TypedDeclaration {

    public ParameterDeclarationPsiNode(@NotNull ASTNode node) {
        super(node);
    }

    public String getTypeText() {
        return getFirstChild().getFirstChild().getText();
    }

}

File: src/main/scala/top/scaleda/verilog/psi/nodes/signal/parameter/ParameterDeclarationPsiNode.scala
package top.scaleda
package verilog.psi.nodes.signal.parameter

import verilog.psi.nodes.signal.SignalDeclarationPsiNode

import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElement
import com.intellij.psi.util.PsiTreeUtil

import java.util

class ParameterDeclarationPsiNode(node: ASTNode) extends SignalDeclarationPsiNode(node) {
  override def getTypeText: String = {
    val typeChildren: Array[PsiElement] =
      util.Arrays.copyOf(this.getChildren, this.getChildren.length - 1) // I don't have range; I don't have semicolon!
    typeChildren.map(_.getText).foldLeft("")(_ + " " + _).trim
  }

  def getListOfParamAssignments: Option[ListOfParamAssignmentsPsiNode] = {
    Option(PsiTreeUtil.getChildOfType(this, classOf[ListOfParamAssignmentsPsiNode]))
  }
}
